components/courses/app/models/courses/progress.rb

```diff
@@ -6,5 +6,9 @@ class Progress < ApplicationRecord
     belongs_to :lecture
     belongs_to :mentor
     belongs_to :homework
+
+    ALLOWED_MARKS = [nil, -1, 0, 0.5, 1, 1.5, 2]
```

Comment:
Why we need `nil` here?

Reply:
In case marks are not set

Reply:
@ivanternovyi what @denysmedynskyi and @nazik23 are saying is that you do not need `nil` in this array, because this:

`validates :homework_mark, inclusion: { in: ALLOWED_MARKS }`

could be turned into:

`validates :homework_mark, inclusion: { in: ALLOWED_MARKS }, allow_blank: true`

(The `allow_blank` takes care of allowing `nil` here, and no need to have it in the array, no need to call `#compact`. Makes for a cleaner code.)

See documentation on `validates` here, including options like `allow_nil` and `allow_blank`: http://api.rubyonrails.org/v5.2.0/classes/ActiveModel/Validations/ClassMethods.html#method-i-validates